File: OC Example/OC Example/ViewController.m
//
//  ViewController.m
//  OC Example
//
//  Created by 任贵权 on 16/5/10.
//  Copyright © 2016年 任贵权. All rights reserved.
//

#import "ViewController.h"
#import "User.h"
#import "UserInfo.h"
#import "ModelUtility/ModelUtility.h"
@import ModelUtility;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    NSDictionary *JSON = @{@"id":@1,@"userInfo":@{@"name":@"isunimp",@"age":@22}};
    User *user = [User mu_modelWithKeyValues:JSON];
    
    NSLog(@"id: %@",user.id);
    NSLog(@"name: %@",user.userInfo.name);
    NSLog(@"age: %@",user.userInfo.age);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swift Examples/Swift Examples/ViewController.swift
//
//  ViewController.swift
//  Swift Examples
//
//  Created by 任贵权 on 16/5/10.
//  Copyright © 2016年 任贵权. All rights reserved.
//

import UIKit
import ModelUtility

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let appDelegate = UIApplication.sharedApplication().delegate as! AppDelegate
        let objectContext = appDelegate.managedObjectContext
        let JSON = ["id":1,"userInfo":["name":"isunimp","age":22]]
        let user =  User.mu_modelWithClass(User.self, keyValues: JSON, inContext: objectContext)
        
        print("id:\(user.id)")
        print("name:\(user.userInfo?.name)")
        print("age:\(user.userInfo?.age)")
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

